src/core_plugins/vega/public/vega_visualization.js

```diff
@@ -1,10 +1,12 @@
 import { Notifier } from 'ui/notify';
 import { VegaView } from './vega_view/vega_view';
 import { VegaMapView } from './vega_view/vega_map_view';
+import { FilterBarQueryFilterProvider } from 'ui/filter_bar/query_filter';
 
-export function VegaVisualizationProvider(vegaConfig, serviceSettings) {
+export function VegaVisualizationProvider(Private, vegaConfig, serviceSettings, indexPatterns, timefilter) {
```

Comment:
get the indexPatterns and timefilter from `vis.API.indexPatterns` and `vis.API.timeFilter`